core/chains/terra/chain.go

```diff
@@ -102,8 +87,36 @@ func (c *chain) MsgEnqueuer() terra.MsgEnqueuer {
 	return c.txm
 }
 
-func (c *chain) Reader() terraclient.Reader {
-	return c.client
+func (c *chain) Reader(name string) (terraclient.Reader, error) {
+	return c.getClient(name)
+}
+
+// getClient returns a client, optionally requiring a specific node by name.
+func (c *chain) getClient(name string) (*terraclient.Client, error) {
+	//TODO cache clients?
```

Comment:
@connorwstein Do you think it is important to cache client instances? Or OK to create new each time? I was thinking that this could stay simple for now, but if we e.g. needed rate limiting then these should be long lived and shared.

Reply:
Long lived isn't necessarily the right way to think about it, since we want to pick up any changes from the DB relatively quickly, but we could maintain an instance per node that we confirm is up to date periodically.

Reply:
Hm the only thing that will change is now we have an instance/connection per job right? So we don't get the http connection caching node wide but that seems ok though since our target scale is only 6 jobs